To: vim_dev@googlegroups.com
Subject: Patch 7.3.415
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.415 (after 7.3.359)
Problem:    Completion of functions stops once a dictionary is encountered.
	    (James McCoy)
Solution:   Return an empty string instead of NULL.
Files:	    src/eval.c


*** ../vim-7.3.414/src/eval.c	2012-01-26 11:45:25.000000000 +0100
--- src/eval.c	2012-01-26 13:45:52.000000000 +0100
***************
*** 21802,21808 ****
  	fp = HI2UF(hi);
  
  	if (fp->uf_flags & FC_DICT)
! 	    return NULL; /* don't show dict functions */
  
  	if (STRLEN(fp->uf_name) + 4 >= IOSIZE)
  	    return fp->uf_name;	/* prevents overflow */
--- 21802,21808 ----
  	fp = HI2UF(hi);
  
  	if (fp->uf_flags & FC_DICT)
! 	    return ""; /* don't show dict functions */
  
  	if (STRLEN(fp->uf_name) + 4 >= IOSIZE)
  	    return fp->uf_name;	/* prevents overflow */
*** ../vim-7.3.414/src/version.c	2012-01-26 13:40:04.000000000 +0100
--- src/version.c	2012-01-26 14:32:16.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     415,
  /**/

-- 
Normal people believe that if it ain't broke, don't fix it.  Engineers believe
that if it ain't broke, it doesn't have enough features yet.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
